ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/bonsai/storage/BonsaiWorldStateKeyValueStorage.java

```diff
@@ -180,16 +177,17 @@ public void clear() {
   }
 
   @Override
-  public FlatDbStrategy getFlatDbStrategy() {
-    return flatDbStrategyProvider.getFlatDbStrategy(composedWorldStateStorage);
+  public BonsaiFlatDbStrategy getFlatDbStrategy() {
+    return (BonsaiFlatDbStrategy)
```

Comment:
suggestion: to avoid having to cast you could make `class FlatDbStrategy` -> `class FlatDbStrategy<T extends FlatDbMode>` or something and then `class BonsaiFlatDbStrategy<FlatDbMode.PARTIAL>`. You could then change `getFlatDbStrategy` return type to `FlatDbStrategy<? extends FlatDbMode>`.
But I'm not sure if that makes sense without getting deep into the code